csvy/readers.py

```diff
@@ -299,3 +300,51 @@ def read_to_list(
             data.append(row)
 
     return data, header
+
+
+def read_to_dict(
+    filename: Path | str,
+    marker: str = "---",
+    encoding: str = "utf-8",
+    csv_options: dict[str, Any] | None = None,
+    yaml_options: dict[str, Any] | None = None,
+    *,
```

Comment:
What does this do? Does it mean `column_names` and `fillvalue` have to be keyword args?

Reply:
Yes. I've opted for this enforcement to clearly distinguish them from the other arguments, which are common for all readers.